From a582d8d6926be45b8e0eea797bfedd9482833274 Mon Sep 17 00:00:00 2001
From: Kavinaya S <kavinaya@codeaurora.org>
Date: Fri, 27 Aug 2021 10:30:10 +0530
Subject: [PATCH] Check for LAYERSERIES_COMPAT

This is to check if LAYERSERIES_COMPAT is set for all
the layers.

Change-Id: I06f8fe59aec73c571391bf7833333e81abbc140f
---
 bitbake/lib/bb/cookerdata.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bitbake/lib/bb/cookerdata.py b/bitbake/lib/bb/cookerdata.py
index 472423f..f671611 100644
--- a/bitbake/lib/bb/cookerdata.py
+++ b/bitbake/lib/bb/cookerdata.py
@@ -403,7 +403,7 @@ class CookerDataBuilder(object):
                     bb.fatal("Layer %s is not compatible with the core layer which only supports these series: %s (layer is compatible with %s)"
                               % (c, " ".join(layerseries), " ".join(compat)))
                 elif not compat and not data.getVar("BB_WORKERCONTEXT"):
-                    bb.warn("Layer %s should set LAYERSERIES_COMPAT_%s in its conf/layer.conf file to list the core layer names it is compatible with." % (c, c))
+                    bb.fatal("Layer %s should set LAYERSERIES_COMPAT_%s in its conf/layer.conf file to list the core layer names it is compatible with." % (c, c))

         if not data.getVar("BBPATH"):
             msg = "The BBPATH variable is not set"
--
2.7.4
